### PR TITLE
Convert Compass to a game HUD module widget

### PR DESCRIPTION
**Pull request by sysrage**
Credit to @reallifegobbo for fixes related to negative facing values.

The old compass module will need to be removed once this is checked in.

### CODE REVIEW

File: game/hud/src/widgets/Compass/index.html [hunk 1]
```diff
@@ -3,16 +3,15 @@
 file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
```
Comment (codecorsair) [inline game/hud/src/widgets/Compass/index.html]: This file serves no purpose here.  delete it

File: game/hud/src/widgets/Compass/index.scss [hunk 2]
```diff
@@ -1,32 +1,25 @@
...
-  font-family: TitilliumBold;
-  src: url(../fonts/TitilliumWeb-Bold.ttf); 
-}
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
 
+$cu-styles-root: '../../../interface-lib/camelot-unchained/styles/';
+@import "camelot-unchained/lib/index.scss";
```
Comment (codecorsair) [inline game/hud/src/widgets/Compass/index.scss]: I don't see any uses of the old cu styles in the compass at all... remove the reference to it.

File: game/hud/src/widgets/Compass/index.scss [hunk 3]
```diff
@@ -1,32 +1,25 @@
...
 
-body {
-  overflow: hidden;
-  cursor: default;
+.compass {
```
Comment (codecorsair) [inline game/hud/src/widgets/Compass/index.scss]: use BEM / OOCSS naming

File: game/hud/src/widgets/Compass/index.scss [hunk 4]
```diff
@@ -1,32 +1,25 @@
...
+  // font-family: TitilliumBold;
```
Comment (codecorsair) [inline game/hud/src/widgets/Compass/index.scss]: delete unused code

File: game/hud/src/widgets/Compass/index.tsx [hunk 5]
```diff
@@ -64,11 +79,20 @@ class Compass extends React.Component<CompassProps, CompassState> {
   }
 
   render() {
+    if (!this.state.visible) return null;
```
Comment (codecorsair) [inline game/hud/src/widgets/Compass/index.tsx]: visibility should be handled by the HUD layout module, not by an individual widget.

File: game/hud/src/widgets/Compass/index.html (hunk 1 above)
Comment (sysrage) [inline game/hud/src/widgets/Compass/index.html]: This should be addressed.

File: game/hud/src/widgets/Compass/index.scss (hunk 2 above)
Comment (sysrage) [inline game/hud/src/widgets/Compass/index.scss]: This should be addressed.

File: game/hud/src/widgets/Compass/index.scss (hunk 3 above)
Comment (sysrage) [inline game/hud/src/widgets/Compass/index.scss]: This should be addressed.

File: game/hud/src/widgets/Compass/index.scss (hunk 4 above)
Comment (sysrage) [inline game/hud/src/widgets/Compass/index.scss]: This should be addressed.

File: game/hud/src/widgets/Compass/index.tsx (hunk 5 above)
Comment (sysrage) [inline game/hud/src/widgets/Compass/index.tsx]: I agree, but this is how all of the widgets are handling visibility at the moment. It's actually going to be useful, to make different parts of a widget visible or not.

**Comment by sysrage** [comment]
I believe all issues have been addressed.

File: game/hud/src/widgets/Compass/index.tsx (hunk 5 above)
Comment (codecorsair) [inline game/hud/src/widgets/Compass/index.tsx]: this is not doing visibility of a part of the widget, this is doing visibility for the entire widget which should be handled externally so we're not even rendering the widget.

**Comment by sysrage** [comment]
Blocked by issue #74.

**Comment by codecorsair** [comment]
#74 is resolved.

**Comment by sysrage** [comment]
Changes from issue #74 have been implemented. I believe PR is ready.

**Pull request merged**